src/multirust-cli/common.rs

```diff
@@ -0,0 +1,134 @@
+//! Just a dumping ground for cli stuff
+
+use multirust::{Cfg, Result, Notification, Toolchain};
+use multirust_utils::{self, utils};
+use multirust_utils::notify::NotificationLevel;
+use std::ffi::OsStr;
+use std::io::{Write, BufRead};
+use std::process::{self, Command};
+use std;
+use tty;
+use term;
+
+pub fn confirm(question: &str, default: bool) -> bool {
+    print!("{} ", question);
+    let _ = std::io::stdout().flush();
+    let input = read_line();
+
+    let r = match &*input {
+        "y" | "Y" => true,
+        "n" | "N" => false,
+        "" => default,
+        _ => false,
+    };
+
+    println!("");
+
+    r
+}
+
+fn read_line() -> String {
+    let stdin = std::io::stdin();
+    let stdin = stdin.lock();
+    let mut lines = stdin.lines();
+    lines.next().unwrap().unwrap()
+}
+
+pub fn set_globals(verbose: bool) -> Result<Cfg> {
+    use download_tracker::DownloadTracker;
+    use std::cell::RefCell;
+
+    let download_tracker = RefCell::new(DownloadTracker::new());
+
+    Cfg::from_env(shared_ntfy!(move |n: Notification| {
+        if download_tracker.borrow_mut().handle_notification(&n) {
+            return;
+        }
+
+        match n.level() {
+            NotificationLevel::Verbose => {
+                if verbose {
+                    verbose!("{}", n);
+                }
+            }
+            NotificationLevel::Info => {
+                info!("{}", n);
+            }
+            NotificationLevel::Warn => {
+                warn!("{}", n);
+            }
+            NotificationLevel::Error => {
+                err!("{}", n);
+            }
+        }
+    }))
+
+}
+
+pub fn run_inner<S: AsRef<OsStr>>(mut command: Command,
```

Comment:
Another nightly feature you may want to give a spin here is to use `CommandExt::exec` on Unix perhaps?